pkg/splunk/enterprise/util_test.go

```diff
@@ -260,4 +270,85 @@ func TestRemoveOwenerReferencesForSecretObjectsReferredBySmartstoreVolumes(t *te
+	// Here in the volume references, secrets splunk-test-sec_1, to splunk-test-sec_4 doesn't exist
+	cr = enterprisev1.ClusterMaster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "idxCluster",
+			Namespace: "testWithNoSecret",
+		},
+		Spec: enterprisev1.ClusterMasterSpec{
+			SmartStore: enterprisev1.SmartStoreSpec{
+				VolList: []enterprisev1.VolumeSpec{
+					{Name: "msos_s2s3_vol", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london", SecretRef: "splunk-test-sec_1"},
+					{Name: "msos_s2s3_vol_2", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london", SecretRef: "splunk-test-sec_2"},
+					{Name: "msos_s2s3_vol_3", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london", SecretRef: "splunk-test-sec_3"},
+					{Name: "msos_s2s3_vol_4", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london", SecretRef: "splunk-test-sec_4"},
+				},
+			},
+		},
+	}
+
+	// S3 secret owner reference removal, with non-existing secret objects
+	err = DeleteOwnerReferencesForS3SecretObjects(client, secret, &cr.Spec.SmartStore)
+	if err == nil {
+		t.Errorf("Should report an error, when the secret object referenced in the volume config doesn't exist")
+	}
+
+	// Smartstore volume config with non-existing secret objects
+	err = DeleteOwnerReferencesForResources(client, &cr, &cr.Spec.SmartStore)
+	if err == nil {
+		t.Errorf("Should report an error, when the secret objects doesn't exist")
+	}
+}
+
+func TestGetSmartstoreRemoteVolumeSecrets(t *testing.T) {
+	cr := enterprisev1.ClusterMaster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "CM",
+			Namespace: "test",
+		},
+		Spec: enterprisev1.ClusterMasterSpec{
+			SmartStore: enterprisev1.SmartStoreSpec{
+				VolList: []enterprisev1.VolumeSpec{
+					{Name: "msos_s2s3_vol", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london", SecretRef: "splunk-test-secret"},
+				},
+			},
+		},
+	}
+
+	client := spltest.NewMockClient()
+
+	// Just to simplify the test, assume that the keys are stored as part of the splunk-test-secret object, hence create that secret object
+	secret, err := splutil.ApplyNamespaceScopedSecretObject(client, "test")
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	_, err = splctrl.ApplySecret(client, secret)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	// Missing S3 access key should return error
+	_, _, _, err = GetSmartstoreRemoteVolumeSecrets(cr.Spec.SmartStore.VolList[0], client, &cr, &cr.Spec.SmartStore)
+	if err == nil {
+		t.Errorf("Missing S3 access key should return an error")
+	}
+
+	secret.Data[s3AccessKey] = []byte("abcdJDckRkxhMEdmSk5FekFRRzBFOXV6bGNldzJSWE9IenhVUy80aa")
+
+	// Missing S3 secret key should return error
+	_, _, _, err = GetSmartstoreRemoteVolumeSecrets(cr.Spec.SmartStore.VolList[0], client, &cr, &cr.Spec.SmartStore)
+	if err == nil {
+		t.Errorf("Missing S3 secret key should return an error")
```

Comment:
Can we have a different message here, easy to debug which condition are we failing at

Reply:
Here, we are filling only the s3AccessKey, and leaving the s3SecretKey missing. Hence the error "Missing S3 secret key should return an error", which is the condition we are testing. There is a comment also added for this test case.